### ISSUE TITLE
Pinterest embed handler URL pattern is incomplete

### ISSUE THREAD

**Issue by dero** [Automattic/amp-wp#5291]
## Bug Description

Many valid Pinterest pin URLs are not captured by the [URL pattern](https://github.com/ampproject/amp-wp/blob/af6f5b7ca20cb16f8795becedfad59c297ea5dc3/includes/embeds/class-amp-pinterest-embed-handler.php#L15
) within `class-amp-pinterest-embed-handler.php`.

Pinterest runs a large number of regional variants of their site. For example, these pins are identical, but hosted on different subdomains:

* https://www.pinterest.com/pin/8092474319950168/
* https://cz.pinterest.com/pin/8092474319950168/
* https://ro.pinterest.com/pin/8092474319950168/

On top of that, some subdomains actually redirect to a different TLD, e.g.:

* https://de.pinterest.com/ --> https://www.pinterest.de/

And there of course is the same pin hosted referenced from the regional TLD as well:

* https://pinterest.de/pin/8092474319950168/

## Expected Behaviour

All valid Pinterest pin URLs are captured and the content is embedded.

## Steps to reproduce

1. Create a new post and add a Custom HTML block.
2. Insert a "regional" Pinterest URL, e. g. `https://cz.pinterest.com/pin/8092474319950168/`
3. Save the post and view it.
4. Notice the embed is not displayed. Instead, the URL is displayed verbatim.
5. Edit the post and change the Custom HTML block contents to `https://pinterest.com/pin/8092474319950168/`.
6. Save the post and view it.
7. Notice the embed is properly displayed.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

**Comment by westonruter** [comment]
I'm surprised that WordPress core does not include Pinterest among [its oEmbed providers](https://github.com/WordPress/wordpress-develop/blob/0013f5979cb00ee9ad3ac5748b72880594bdbb1c/src/wp-includes/class-wp-oembed.php#L53-L115).

We can include this fix in 2.0.1.

**Comment by pierlon** [comment]
Hi @dero, can you please verify that the issue is fixed by testing the build linked here: https://github.com/ampproject/amp-wp/pull/5299#issuecomment-683334330.

**Comment by dero** [comment]
Hi @pierlon – thank you, I have verified those changes address issues outlined in this ticket. 👍 

However, I'm also going to follow up on the PR with a couple comments.

**Comment by westonruter** [comment]
**QA Passed**

Testing with this content:

![image](https://user-images.githubusercontent.com/134745/92176885-3c700980-edf4-11ea-8069-aaeb651baca7.png)

Before | After
-------|------
![image](https://user-images.githubusercontent.com/134745/92176966-6b867b00-edf4-11ea-9aa6-43434d2c3ad0.png) | ![image](https://user-images.githubusercontent.com/134745/92176951-5f9ab900-edf4-11ea-9aa0-3a2c6929ec96.png)

**Comment by westonruter** [comment]
In testing I noticed the Pins do not render well. I've worked out an improved approach to rendering that ensures the correct height is applied to `amp-pinterest` by using Pinterest REST API call. This is proposed for the Jetpack Pinterest block, but we should port it into the AMP plugin: https://github.com/Automattic/jetpack/pull/17086

Aside: I wonder why Jetpack doesn't register an oEmbed handler so that the Embed block can't be used.
